src/datasources/cache/cache.first.data.source.spec.ts

```diff
@@ -35,11 +36,78 @@ describe('CacheFirstDataSource', () => {
     );
   });
 
-  it('should return the data returned by the underlying network interface', async () => {
+  afterEach(() => {
+    jest.useRealTimers();
+  });
+
+  it('should return the data returned by the underlying network interface and cache it', async () => {
+    const targetUrl = faker.internet.url({ appendSlash: false });
+    const cacheDir = new CacheDir(faker.word.sample(), faker.word.sample());
+    const notFoundExpireTimeSeconds = faker.number.int();
+    const data = JSON.parse(fakeJson());
+    mockNetworkService.get.mockImplementation((url) => {
+      switch (url) {
+        case targetUrl:
+          return Promise.resolve({ data });
+        default:
+          return Promise.reject(`No matching rule for url: ${url}`);
+      }
+    });
+
+    const actual = await cacheFirstDataSource.get({
+      cacheDir,
+      url: targetUrl,
+      notFoundExpireTimeSeconds,
+    });
+
+    expect(actual).toEqual(data);
+    expect(mockNetworkService.get).toHaveBeenCalledTimes(1);
+    expect(fakeCacheService.keyCount()).toBe(1);
```

Comment:
Nit: can we add a comment here clarifying that the invalidation key has yet to be added.

Reply:
I've added several comments explaining the test expectations in [f895765](https://github.com/safe-global/safe-client-gateway/pull/970/commits/f89576594ef369e007eac0e242330f957c05b3b7), let me know if it's not very understandable, or you want to write it differently 🙂